reports/facter.go

```diff
@@ -0,0 +1,51 @@
+package reports
+
+import (
+	"encoding/json"
+	"os/exec"
+
+	"github.com/pkg/errors"
+)
+
+// PuppetFacts stores output of puppet facts command
+type PuppetFacts map[string]interface{}
+
+// GetPuppetFacts will exec into the PuppetFacts interface
+func GetPuppetFacts() (PuppetFacts, error) {
+	cmd := exec.Command("puppet", "facts")
+
+	// cmd.Stderr = os.Stderr
+	o, err := cmd.Output()
+	if err != nil {
+		return PuppetFacts{}, errors.Wrap(err, "exec puppet facts")
+	}
+
+	var pf PuppetFacts
+
+	if err := json.Unmarshal(o, &pf); err != nil {
+		return PuppetFacts{}, errors.Wrap(err, "failed unmarshalling Puppet Facts")
+	}
+
+	v := pf["values"].(map[string]interface{})
+
+	pf = Flatten(v)
+
+	return pf, nil
+}
+
+// Flatten takes a map[string]interface and flattens it
```

Comment:
What are you flattening the results to? Are they always strings?

Reply:
well.  in this case they are all strings :) but would like to reflect type.